.github/workflows/run_test_suite.yaml

```diff
@@ -82,7 +82,24 @@ jobs:
         with:
           command: test
           args: --features test_kubo,headers
-
+  run-test-suite-linux-c:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: Swatinem/rust-cache@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          override: true
+          profile: minimal
+          toolchain: stable
+      - name: 'Install environment packages'
+        run: |
+          sudo apt-get update -qqy
+          sudo apt-get install jq protobuf-compiler cmake
+      - name: 'Run C tests'
```

Comment:
Can we also build the example as a step in the test run? This will make sure our example doesn't fall out of date (kind of like a doc test of sorts).

Reply:
In `run-test-suite-linux`? Ya, but then I don't understand what this job is providing (and could just be thought of as integration tests for the rust FFI layer, rather than a C environment), to just be a part of e.g. `run-test-suite-linux`. As of now the example IS the test

Reply:
No, I want you to do it here as a step before or after the test run.

Reply:
I assume that the example has the same binary dependencies as the tests.

Reply:
Ah I gotcha now, split up the build from the run in their own steps, sg!